### PR TITLE
[Tooling] Create a Single, Unified GitHub Releases for WP Alpha, WP Beta & JP Beta

### PR DESCRIPTION
**Pull request by AliSoftware**
This is part of project paaHJt-2s9-p2

ℹ️  This PR targets `release/18.2` in the hope I get the occasion to use it in this release in case it's merged in time.

## What it does

Now that https://github.com/wordpress-mobile/WordPress-Android/pull/15273 has been merged and the 3 Release Builds (WP Alpha, WP Beta, JP Beta) are done in parallel, this PR is now able to **create a single GitHub Release for all 3 builds**, instead of creating one GitHub Release for each like before.

## How

 - Refactor the `create_gh_release` lane to be able to include multiple assets for multiple apps and versions at once, merging release notes as necessary into one description.
 - Update the YARD doc of `build_*` lanes to mention the `app:` parameters and update their call to `create_gh_release`
 - Update the `.circleci/config.yml` file to call `create_gh_release` with the 3 build artifacts only created once the 3 parallel builds have finished.

## To Test

* Run `git checkout -b ci-test/pr15319 origin/jetpack-infra/unified-gh-release^1` to cut a new branch from the commit **before the tip of this one**.
  * This will include the debug commit 1e5c7795f269cb7e340bb9405dadd11839d59b4f – which changes the `versionName` to a dummy `18.2-rc-999`, and disables `upload_to_playstore:true` and the `slack` action – but not include the last commit of this branch which reverts those debug changes.
   * This is done to avoid noise during testing, but also to ensure the test GitHub Release gets a new name (instead of being auto-matched with existing tag `18.2-rc-3`), and so that the builds won't fail with "versionCode already exists in PlayStore"
* Push your new test branch to origin: `git push -u origin ci-test/pr15319`
* Trigger a release build: `bundle exec fastlane trigger_release_build branch_to_build:ci-test/pr15319`
* Wait for the CI workflow to finish, then check that a single, unified [GitHub Release](https://github.com/wordpress-mobile/WordPress-Android/releases) draft has been created.

The generated GitHub Release draft should:
 - Point to the commit sha1 of your `ci-test/pr15319` branch, which should be `1e5c7795f269cb7e340bb9405dadd11839d59b4f`
 - Be titled `"18.2-rc-999"`
 - Contain BOTH the WordPress and Jetpack release notes or 18.2 – as found in `WordPress/{metadata,jetpack_metadata}/release_notes.txt` files
 - Contain the 3 `.aab` artefacts (WP Alpha, WP Beta, JP Beta)
 - Be marked as "This is a pre-release"

<details><summary>Screenshot of the expected GitHub Release draft</summary>

</details>

#### Cleanup

* Delete your `ci-test/pr15319` branch from origin.
* Delete the draft GitHub Release named `18.2-rc-999`.

### CODE REVIEW

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: Would this shorter name work, too?

```suggestion
    tmp_file = File.absolute_path('unified-release-notes.txt')
```

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]:
```suggestion
  #   - If `prerelease:<true|false>` is not provided, the pre-release status will be guessed from the version name
```

**Review by mokagio — APPROVED** [review]
This is very neat. Blown away by the speed at which CI built. I went on the tab to check on the progress and it had already finished! 🤯 

I verified the generated release matches the expectation 👌 

Not sure what your plans for merging are. In case you haven't thought of it already, I'd suggest rewriting the history to `drop` the last two commits.

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]: uhu?  Not sure why I went with that one 🤦 I guess it's because I refactored that part of the code multiple times and ended up not realizing what the end name was 😅  I will definitively update the variable name with your suggestion.

Comment (AliSoftware) [reply fastlane/Fastfile]: Oh I remember now: at first `tmp_file` was a `IO::File`, so when I switched the variable to be just the path, to avoid confusion with past code I wanted to renaming it to `tmp_file_path`. And because I was tired, I wrote `tmp_file_file` instead of `tmp_file_path`… 🤦 

Anyway, fixed in f2a3e704a6eb47da9d58ff527cca6b89e01c0cb4

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]: Fixed in f2a3e704a6eb47da9d58ff527cca6b89e01c0cb4; also changed the wording from "guessed" to "inferred" (as that was what I originally meant, but couldn't remember the word yesterday 😄)

Comment (AliSoftware) [reply fastlane/Fastfile]: > This is very neat. Blown away by the speed at which CI built. I went on the tab to check on the progress and it had already finished! 🤯
 
Yep, definitively! Having this project done (paaHJt-2s8-p2 + paaHJt-2s9-p2) will change my life as Release Manager!! I'm so happy for this change, no more spending the whole morning waiting for CI (and having to wait for WP CI to finish before I need to trigger JP CI) and no more having to do half of the tasks by hand anymore 🎉  #relief

> Not sure what your plans for merging are. In case you haven't thought of it already, I'd suggest rewriting the history to `drop` the last two commits.

Yeah I did it that way because last time I used a debug commit like that, [the PR was accidentally merged before we pushed back the revert commit](https://github.com/wordpress-mobile/WordPress-Android/pull/15273#issuecomment-915084562), so I figured it was better to include the debug commit and its revert this time, and ask you to point before the revert commit. 

That being said, I think I'll keep those 2 commits in the merge, because rewriting history to drop them would make the PR description invalid if I did, and it can still make sense to keep track on how the PR was tested. Not a big deal once the PR is merged you'd say, and I can also see the value in keeping the git history cleaner, but I still like to keep consistency in the PR comments and desc even post-merge, in case we need to come back to this PR later one day and need to make sense of it 🙃

**Comment by AliSoftware** [comment]
> This is very neat. Blown away by the speed at which CI built. I went on the tab to check on the progress and it had already finished! 🤯
 
Yep, definitively! Having this project done (paaHJt-2s8-p2 + paaHJt-2s9-p2) will change my life as Release Manager!! I'm so happy for this change, no more spending the whole morning waiting for CI (and having to wait for WP CI to finish before I need to trigger JP CI) and no more having to do half of the tasks by hand anymore 🎉  #relief

> Not sure what your plans for merging are. In case you haven't thought of it already, I'd suggest rewriting the history to `drop` the last two commits.

Yeah I did it that way because last time I used a debug commit like that, [the PR was accidentally merged before we pushed back the revert commit](https://github.com/wordpress-mobile/WordPress-Android/pull/15273#issuecomment-915084562), so I figured it was better to include the debug commit and its revert this time, and ask you to point before the revert commit. 

That being said, I think I'll keep those 2 commits in the merge, because rewriting history to drop them would make the PR description invalid if I did, and it can still make sense to keep track on how the PR was tested. Not a big deal once the PR is merged you'd say, and I can also see the value in keeping the git history cleaner, but I still like to keep consistency in the PR comments and desc even post-merge, in case we need to come back to this PR later one day and need to make sense of it 🙃

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile]:
```suggestion
  # bundle exec fastlane create_gh_release version:{name:12.3-rc-4} prerelease:true             # Includes existing assets for WPBeta+JPBeta 12.3-rc-4
```

**Pull request merged**
